app/routes.py

```diff
@@ -1,2 +1,299 @@
+        return make_response(jsonify(tasks_response), 200)  
+    
+    if sort_query == "asc":
+
+        ordered_query = Task.query.order_by(Task.title.asc())
+        
+    elif sort_query == "desc":
+
+        ordered_query = Task.query.order_by(Task.title.desc())
+    
+    elif sort_query == "id":
+
+        ordered_query = Task.query.order_by(Task.id)
+
+    for task in ordered_query:
+        sorted_response.append(task.create_json())
+    
+    return jsonify(sorted_response), 200
+
+
+        
+@tasks_bp.route("", methods=["POST"], strict_slashes=False)
+def create_new_task():
+
+    request_body = request.get_json()
+    if not request_body or not request_body.get("title") or not request_body.get("description") or "completed_at" not in request_body:
+        return make_response({"details": "Invalid data"}, 400)
+    
+    new_task = Task.from_json(request_body)
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return make_response({"task": new_task.create_json()}, 201)
+
+
+@tasks_bp.route("/<task_id>", methods=["GET"], strict_slashes=False)
+def get_task_by_id(task_id):
+    request_body = request.get_json()
+    task = Task.query.get(task_id)
+
+    if task is None:
+            return make_response("", 404)
+
+    if task.goal_id is None:
+        return make_response({"task": task.create_json()}, 200)
+    
+    else:
+    
+        return make_response({"task": task.create_json_with_goal_id()}, 200)      
+
+
+@tasks_bp.route("/<task_id>", methods=["PUT"], strict_slashes=False)
+def update_task(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+            return make_response("", 404)
+
+    else:
+        form_data = request.get_json()
+
+        task.title = form_data["title"]
+        task.description = form_data["description"]
+        task.completed_at = form_data["completed_at"]
+
+        db.session.commit()
+        return make_response({"task": task.create_json()}, 200)
+
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"], strict_slashes=False)
+def delete_task(task_id):   
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return make_response("", 404)
+
+    else:
+        db.session.delete(task)
+        db.session.commit()
+
+    return make_response({"details":f"Task {task.id} \"{task.title}\" successfully deleted"}, 200)
+
+
+@tasks_bp.route("<task_id>/mark_complete", methods=["PATCH"], strict_slashes=False)
+def mark_complete(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+            return make_response("", 404)
+
+    task.completed_at = datetime.utcnow()
+    db.session.commit()
+
+    SLACK_TOKEN = os.environ["SLACK_API_TOKEN"]
+    slack_channel = "task-notifications"
+    slack_icon_url = "https://slack.com/api/chat.postMessage"
+    text = f"Someone just completed the task {task.title}"
+
+    response = requests.post('https://slack.com/api/chat.postMessage', {
+        'token': slack_token,
+        'channel': slack_channel,
```

Comment:
You forgot to capitalize this.  So some tests are failing. 

```suggestion
        'channel': SLACK_TOKEN,
```